analytics/tests/test_views.py

```diff
@@ -233,6 +251,28 @@ def test_analytics_not_running(self) -> None:
                                      {'chart_name': 'number_of_humans'})
         self.assert_json_error_contains(result, 'No analytics data available')
 
+    def test_get_chart_data_for_realm(self) -> None:
+        user_profile = self.example_user('hamlet')
+        self.login(user_profile.email)
+
+        result = self.client_get('/json/analytics/chart_data/realm/zulip/',
+                                 {'chart_name': 'number_of_humans'})
+        self.assert_json_error(result, "Must be an server administrator", 400)
```

Comment:
@timabbott This already tests the new decorator function( just like `require_realm_admin`, which is also tested by similar requests in other tests). 
Actually I give this a shot, and it is really a pain to add a test for this, the problem here is that if we want to test the decorator (in its own commit) we have to mock `zulip_login_required` decorator itself and from my search mocking a decorator is a complex task which become more worse here as we are further testing another decorator (because decorators act at the load time rather than run time which makes it difficult). Another alternative here is that to mock whole request object, but this approach doesn't look right to me because of redefining many functions and other object members. 
If we still want to add test, can you please help me here?

Reply:
OK, I guess the current tests are fine.